### PR TITLE
bash-config: /etc/profile.d cleanup survey

### PR DESCRIPTION
**Pull request by leedagee**
Topic Description
-----------------

[bash-config#23](https://github.com/AOSC-Dev/bash-config/pull/23) fixed a bash initialization related bug, while it reveals a wide range of flaws in our packaging work. This topic moves the misused profile.d scripts to correct directories for a consistent behavior.

Package(s) Affected
-------------------

Currently, these package provides /etc/profile.d:

- [x] ~32subsystem~
- [x] ~apache-ant~
- [x] ~apache-maven~
- [x] ~arduino-mk~
- [x] autojump-rs
- [x] ~bash-completion~
- [x] ~bash-startup~
- [x] ~byobu~
- [x] ~cuda~
- [x] dotnet-host
- [x] ~elfutils~
- [x] ~firefox~
- [x] ~flatpak~
- [x] fzf
- [x] ~gawk~
- [x] gpm (help wanted, seems ok)
- [x] ~gradle~
- [x] grc (placed in profile.d by upstream, the broken feature is not widely used, moved to bashrc.d)
- [x] ~hook-uname~ (~help wanted, why do oma provides output this~, /etc/profile.d is dropped by 68ec4845b)
- [x] ~ksshaskpass~
- [x] ~libreoffice~
- [x] ~mercurial~
- [x] ~midori~
- [x] ~mingw+w64~
- [x] ~mozilla-common~
- [x] nvm
- [x] oma (invalid, discuss in oma topic)
- [x] ~openjdk~
- [x] ~openjdk-11~
- [x] ~openjdk-8~
- [x] ~perl~
- [x] powershell
- [x] ~qt-5~
- [x] ruby
- [x] ~seahorse~
- [x] ~snapd~
- [x] ~stackless-env~ (~help wanted~, drop orphaned)
- [x] ~thunderbird~
- [x] udisks 
- [x] vte
- [x] wezterm
- [x] ~x11-ssh-askpass~

Overall, changes actually applied for:
- udisks
- ~powershell~
- autojump-rs
- fzf
- grc
- wezterm
- dotnet-host
- nvm
- ruby
- vte

(Strike-through means the package uses /etc/profile.d correctly, this survey topic is not applicable.)

Security Update?
----------------

No 

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: lang-js/nvm/autobuild/build [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 EOF
 cat aosc_nvm_stub.sh nvm.sh > nvm_aosc.sh
 
-abinfo "Installing nvm profile, bash completion and doc..."
-install -Dm644 nvm_aosc.sh "${PKGDIR}/etc/profile.d/30-nvm.sh"
-install -Dm644 bash_completion "${PKGDIR}/usr/share/bash-completion/completions/nvm"
-install -d "${PKGDIR}/usr/share/doc/nvm/"
-cp *.md "${PKGDIR}/usr/share/doc/nvm/"
+abinfo "Installing nvm shell startup files, bash completion and doc..."
+install -Dvm644 nvm_aosc.sh "${PKGDIR}/etc/bashrc.d/30-nvm.sh"
```
Comment (cthbleachbit) [inline lang-js/nvm/autobuild/build]: Any rationale on why this export should be a bash specifc thing? Looks like NVM_DIR should be set regardless of default shell choice.

Also note that desktop environments generally don't source `/etc/bashrc.d` but would include `/etc/profile.d`. If this environment variable were to be used by any graphical applications (say, an IDE or a graphical text editor) they will no longer see `NVM_DIR` in their environment.

File: lang-js/nvm/autobuild/build [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+install -Dvm644 bash_completion "${PKGDIR}/usr/share/bash-completion/completions/nvm"
+install -dv "${PKGDIR}/etc/zsh/zshrc.d"
+ln -vs ../../bashrc.d/30-nvm.sh "${PKGDIR}/etc/zsh/zshrc.d/30-nvm"
```
Comment (cthbleachbit) [inline lang-js/nvm/autobuild/build]: Just leave it in `/etc/profile.d`...

**Review by cthbleachbit — DISMISSED** [review]

File: lang-js/nvm/autobuild/build [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-cat aosc_nvm_stub.sh nvm.sh > nvm_aosc.sh
-
-abinfo "Installing nvm profile, bash completion and doc..."
-install -Dm644 nvm_aosc.sh "${PKGDIR}/etc/profile.d/30-nvm.sh"
-install -Dm644 bash_completion "${PKGDIR}/usr/share/bash-completion/completions/nvm"
-install -d "${PKGDIR}/usr/share/doc/nvm/"
-cp *.md "${PKGDIR}/usr/share/doc/nvm/"
+abinfo "Creating symlink to nvm.sh in bashrc.d and zshrc.d"
+mkdir -pv "${PKGDIR}"/etc/{bashrc.d,zsh/zshrc.d}
+ln -vs /usr/share/nvm/nvm.sh "${PKGDIR}/etc/bashrc.d/30-nvm"
```
Comment (MingcongBai) [inline lang-js/nvm/autobuild/build]: Use relative paths here.

File: lang-js/nvm/autobuild/build [hunk 4]
```diff
@@ -1,13 +1,10 @@
-abinfo "Creating nvm profile..."
+abinfo "Installing nvm.sh, bash completion and doc..."
+install -Dvm644 nvm.sh "${PKGDIR}/usr/share/nvm/nvm.sh"
```
Comment (MingcongBai) [inline lang-js/nvm/autobuild/build]: Missing absolute path "$SRCDIR"/...

File: lang-js/nvm/autobuild/build [hunk 5]
```diff
@@ -1,13 +1,10 @@
...
+install -Dvm644 bash_completion "${PKGDIR}/usr/share/bash-completion/completions/nvm"
+mkdir -pv "${PKGDIR}/usr/share/doc/nvm/"
+cp -v *.md "${PKGDIR}/usr/share/doc/nvm/"
```
Comment (MingcongBai) [inline lang-js/nvm/autobuild/build]: ... Until here.

Comment (MingcongBai) [reply lang-js/nvm/autobuild/build]: Also, you could have just used `install -Dvm644 "$SRCDIR"/*.md -t "$PKGDIR"/usr/share/doc/nvm/`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by leedagee** [comment]
powershell is cherry picked from openssl-3.3
wezterm is left unchanged

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**